resources-build-logic/src/main/kotlin/multiplatform-library-convention.gradle.kts

```diff
@@ -52,7 +52,6 @@ kotlin {
         val macosX64Main by getting
 
         val macosMain by creating {
-            dependsOn(commonMain)
```

Comment:
This was conflicting with the structure in the `apple-main-convention` module, leading to the IDE properly resolving to `appleMain` but causing the compiler to fail.